lockup: add vesting info, add graphs with schedule to the doc

telezhnaya opened the pull request:
Let's review the contents, then we will ask the designer to re-draw the pictures.

I added info about vesting, reordered things a little, added graphs.

UPD: Here is a direct link to the updated documentation page for your convenience: https://docs-pr-691.onrender.com/docs/tokens/lockup

telezhnaya commented:
@eriktrautman I am updating technical documentation for our lockup contract
https://github.com/near/core-contracts/pull/144

I decided that it's crucial to update high-level doc as well. All the comments are highly appreciated.

frol commented on docs/tokens/lockup.md:
```diff
@@ -20,42 +21,83 @@ Another factor in lockups is the "cliff", which means that no tokens are unlocke
 *See the FAQ at the end for questions*
 
 
+## The Lockup Contract at Near
```
```suggestion
## The Lockup Contract at NEAR
```

frol commented on docs/tokens/lockup.md:
```diff
@@ -20,42 +21,83 @@ Another factor in lockups is the "cliff", which means that no tokens are unlocke
+[Lockup](https://github.com/near/core-contracts/tree/master/lockup) is a special smart contract that ensures that the full or the partial amount is not transferable until it is supposed to be.
+
+The lockups are implemented as a separate smart contract from your main account.
+Thus, if you have received tokens prior to [Phase 2](https://near.org/blog/near-mainnet-phase-2-unrestricted-decentralized/), you will get two things:
```
```suggestion
Thus, if you have received tokens prior to [Phase II](https://near.org/blog/near-mainnet-phase-2-unrestricted-decentralized/), you will get two things:
```

frol commented on docs/tokens/lockup.md:
```diff
@@ -20,42 +21,83 @@ Another factor in lockups is the "cliff", which means that no tokens are unlocke
+<img width="563" alt="Screenshot 2021-05-07 at 14 21 34" src="https://user-images.githubusercontent.com/11246099/117442575-a28ecc80-af3f-11eb-8304-01ce5f2eaa5c.png">
+
+[deprecated] Apart from the lockup timestamp, there is a lockup duration.
+`lockup_duration` is the interval between [the Phase 2 launch](https://near.org/blog/near-mainnet-phase-2-unrestricted-decentralized/) (October 14th) and the moment when tokens start to unlock.
```
```suggestion
`lockup_duration` is the interval between [the Phase II launch](https://near.org/blog/near-mainnet-phase-2-unrestricted-decentralized/) (October 14th) and the moment when tokens start to unlock.
```

frol reviewed (approved):
It looks more accurate and specific to me now!

My nits are just about "Phase 2" being usually spelled as "Phase II" in our posts.

telezhnaya commented:
@frol @eriktrautman I added pictures that Jake sent me. Please have a look.

I also fixed Phase II date, it was the evening of Oct 13. Now it's correct both in text and on pictures.

frol commented on docs/tokens/lockup.md:
```diff
@@ -20,42 +21,83 @@ Another factor in lockups is the "cliff", which means that no tokens are unlocke
+Thus, if you have received tokens prior to [Phase II](https://near.org/blog/near-mainnet-phase-2-unrestricted-decentralized/), you will get two things:
+
+1. A regular account (also called "Owner Account" in context of lockups), let's say `user.near` or `3e52c197feb13fa457dddd102f6af299a5b63465e324784b22aaa7544a7d55fb`;
+2. A lockup contract, with name `4336aba00d32a1b91d313c81e8544ea1fdc67284.lockup.near`.
```
```suggestion
2. A lockup contract, with a name like `4336aba00d32a1b91d313c81e8544ea1fdc67284.lockup.near`.
```

frol commented on docs/tokens/lockup.md:
```diff
@@ -20,42 +21,83 @@ Another factor in lockups is the "cliff", which means that no tokens are unlocke
+1. A regular account (also called "Owner Account" in context of lockups), let's say `user.near` or `3e52c197feb13fa457dddd102f6af299a5b63465e324784b22aaa7544a7d55fb`;
+2. A lockup contract, with name `4336aba00d32a1b91d313c81e8544ea1fdc67284.lockup.near`.
+
+The owner account is created first, either by following the NEAR Drop process or by creating new key pair using Trust, Ledger or another wallet.
```
```suggestion
The owner account is created first, either by following the NEAR Drop process or by creating a new key pair using Trust, Ledger, or another wallet.
```

frol commented on docs/tokens/lockup.md:
```diff
@@ -20,42 +21,83 @@ Another factor in lockups is the "cliff", which means that no tokens are unlocke
+
+The lockup contract is then deployed with a predictable name.
+It is defined as `hash(owner_account_id)[:20]` encoded in `hex` and deployed as subaccount under `lockup.near`.
+It means that all lockup contracts end with `.lockup.near`.
```
```suggestion
It means that all lockup contracts are deployed to accounts that are named with `.lockup.near` at the end.
```

frol commented on docs/tokens/lockup.md:
```diff
@@ -20,42 +21,83 @@ Another factor in lockups is the "cliff", which means that no tokens are unlocke
+It is defined as `hash(owner_account_id)[:20]` encoded in `hex` and deployed as subaccount under `lockup.near`.
+It means that all lockup contracts end with `.lockup.near`.
+
+If you have added or earned additional tokens above the initial balance, they are considered unlocked and can be freely withdrawn.
```
```suggestion
If you have received additional tokens to the lockup account, they are considered unlocked and can be freely withdrawn.
```

frol commented on docs/tokens/lockup.md:
```diff
@@ -20,42 +21,83 @@ Another factor in lockups is the "cliff", which means that no tokens are unlocke
+It means that all lockup contracts end with `.lockup.near`.
+
+If you have added or earned additional tokens above the initial balance, they are considered unlocked and can be freely withdrawn.
+For example, any rewards that are earned using this lockup contract (eg from delegation) or any other funds sent to this lockup contract can be withdrawn by the owner at any time.
```
```suggestion
For example, any rewards that are earned using this lockup account (e.g. from delegation) or any other funds sent to this lockup contract can be withdrawn by the owner at any time.
```

frol commented on docs/tokens/lockup.md:
```diff
@@ -20,42 +21,83 @@ Another factor in lockups is the "cliff", which means that no tokens are unlocke
+2. Cliff vesting period.
 
-`Lockup` is a special smart contract that ensures that the full amount or even a partial amount is not transferable until it is supposed to be.
+Vesting process contains of 3 timestamps: `start_date`, `cliff_date`, `end_date`.
```
```suggestion
Vesting process is defined by 3 timestamps: `start_date`, `cliff_date`, `end_date`.
```

telezhnaya replied:
Changed it to `includes` because `is defined by` sounds for me like if I have these 3 timestamps, I should be able to say which vesting process it is.

frol commented on docs/tokens/lockup.md:
```diff
@@ -20,42 +21,83 @@ Another factor in lockups is the "cliff", which means that no tokens are unlocke
+![lockup_4](/docs/assets/lockup/lockup_4.png)
 
-Lockup contract is then deployed with predictable name. It is defined as `hash(owner_account_id)[:20]` encoded in `hex` and deployed as subaccount under `lockup.near`. Which means that all lockup contracts end with `.lockup.near`.
+The resulting graph of the tokens becoming liquid is always the minimum between lockup and vesting mechanics.
```
```suggestion
The liquid tokens balance is always the minimum between unlocked and vested values.
```

frol commented on docs/tokens/lockup.md:
```diff
@@ -20,42 +21,83 @@ Another factor in lockups is the "cliff", which means that no tokens are unlocke
 
-When the lockup has been fully released (lockup start date + release length has passed), the Owner Account can call to terminate the lockup and withdraw all the funds from it to the owner account.
+At the moment of termination, we fix the number of tokens that should be finally unlocked at the end.
+We continue to unlock the tokens as we normally do that by getting the minimum between lockup and vesting graph.
```
```suggestion
We continue to unlock the tokens as we normally do that by getting the minimum between unlocked and vested amounts.
```

frol commented on docs/tokens/lockup.md:
```diff
@@ -20,42 +21,83 @@ Another factor in lockups is the "cliff", which means that no tokens are unlocke
+![lockup_5](/docs/assets/lockup/lockup_5.png)
 
-When the lockup has been fully released (lockup start date + release length has passed), the Owner Account can call to terminate the lockup and withdraw all the funds from it to the owner account.
+At the moment of termination, we fix the number of tokens that should be finally unlocked at the end.
```
```suggestion
At the moment of termination, we stop the vesting process, so the vested amount is going to remain constant after that; the lockup process keeps going and will unlock the tokens on its schedule.
```

frol commented on docs/tokens/lockup.md:
```diff
@@ -20,42 +21,83 @@ Another factor in lockups is the "cliff", which means that no tokens are unlocke
-When the lockup has been fully released (lockup start date + release length has passed), the Owner Account can call to terminate the lockup and withdraw all the funds from it to the owner account.
+At the moment of termination, we fix the number of tokens that should be finally unlocked at the end.
+We continue to unlock the tokens as we normally do that by getting the minimum between lockup and vesting graph.
+We stop when we reach the desired amount of tokens.
```
This sentence is unclear :thinking:

telezhnaya replied:
Removed it, it does not give any value here
